### ISSUE TITLE
Overriding global jsx prop is messing with typescript

### ISSUE THREAD

**Issue by JClackett** [algolia/ui-components#28]
https://github.com/algolia/ui-components/blob/58ee539ae56cb27c9662f341a04dcc06f85e554a/packages/shared/src/Renderer.ts#L12

This is overriding the key prop in my application and isn't aligned with the react types, which accept a bigint. so when spreading props which may contain a key property im getting type errors across my entire application

```
 Types of property 'key' are incompatible.
      Type 'Key | null | undefined' is not assignable to type 'string | number | null | undefined'.
        Type 'bigint' is not assignable to type 'string | number | null | undefined'.ts(2322)
```

**Comment by JClackett** [comment]
Is it needed to override the global namespace and if so, can it align with what react accepts

**Comment by JanMisker** [comment]
I came here exactly for this. I also cannot re-override the faulty declaration, or am I missing something? @JClackett you found a solution already?

**Comment by JClackett** [comment]
> I came here exactly for this. I also cannot re-override the faulty declaration, or am I missing something? @JClackett you found a solution already?

I'm using pnpm and did a 'patch package' to the underlying types to add bigint

**Comment by JanMisker** [comment]
thanks for the tip, also had another package with an error in it

**Comment by Haroenv** [comment]
Just to confirm, does https://github.com/algolia/ui-components/pull/29 work for you?

**Comment by Haroenv** [comment]
this is now released in 1.2.2, fix on InstantSearch coming soon
